src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/DoNotDisableHttpClientCRLCheck.cs

```diff
@@ -235,8 +235,8 @@ public override void Initialize(AnalysisContext context)
                             }
                             finally
                             {
-                                rootOperationsNeedingAnalysis.Free();
-                                allResults?.Free();
+                                rootOperationsNeedingAnalysis.Free(compilationAnalysisContext.CancellationToken);
+                                allResults?.Free(compilationAnalysisContext.CancellationToken);
```

Comment:
`allResults` is local to this method, so it can always be returned when the method completes.

```suggestion
                                allResults?.Free();
```

Reply:
I kept it consistent with the other free call in the finally, just feels more cleaner.